include/openenclave/bits/types.h

```diff
@@ -80,13 +80,17 @@ typedef long long ptrdiff_t;
 
 /**
  * This enumeration defines values for the **enclave_type** parameter
- * passed to **oe_create_enclave()**. Currently only **OE_ENCLAVE_TYPE_SGX**
- * is supported.
+ * passed to **oe_create_enclave()**. OE_ENCLAVE_TYPE_AUTO will pick the type
+ * based on the target platform that is being built, such that x64 binaries will
+ * use SGX.
+ * **OE_ENCLAVE_TYPE_SGX** will force the platform to use SGX, but any platform
+ * other than x64 will not support this and will generate errors.
  */
 typedef enum _oe_enclave_type
 {
-    OE_ENCLAVE_TYPE_UNDEFINED,
+    OE_ENCLAVE_TYPE_AUTO,
     OE_ENCLAVE_TYPE_SGX,
+    OE_ENCLAVE_TYPE_UNDEFINED,
```

Comment:
When would an app use UNDEFINED instead of AUTO?   Trying to figure out why it would have a different value.

Reply:
It should never be used. It is an invalid value. I am open to renaming this if there is a better name.

Reply:
If it should never be used, and you're making a breaking change anyway, then it should simply be removed from the enum, not added as another new value that should never be used.

Reply:
The other alternative is to leave it as a deprecated alias for AUTO, along the lines of
#ifdef OE_V1_APIS
# define OE_ENCLAVE_TYPE_UNDEFINED OE_ENCLAVE_TYPE_AUTO
#endif